src/engine/strat_engine/pool.rs

```diff
@@ -127,9 +161,25 @@ impl StratPool {
 
 impl Pool for StratPool {
     fn create_filesystems<'a, 'b>(&'a mut self,
-                                  _specs: &[&'b str])
+                                  specs: &[&'b str])
                                   -> EngineResult<Vec<(&'b str, Uuid)>> {
-        unimplemented!()
+        let names = BTreeSet::from_iter(specs);
+        for name in names.iter() {
+            if self.filesystems.contains_name(name) {
+                return Err(EngineError::Engine(ErrorEnum::AlreadyExists, name.to_string()));
+            }
+        }
+
+        let mut result = Vec::new();
+        for name in names.iter() {
+            let uuid = Uuid::new_v4();
+            let dm = try!(DM::new());
+            let new_filesystem = try!(StratFilesystem::new(uuid, name, &dm, &mut self.thin_pool));
+            self.filesystems.insert(new_filesystem);
+            result.push((**name, uuid));
```

Comment:
Might name only need one dereference operator, ```*name``` instead of ```**name```? I can't figure out why it should be two.